### ISSUE TITLE
HTTP 400 Bad Request

### ISSUE THREAD

**Issue by MegaS0ra** [cweiske/grauphel#25]
Hi,

I have a problem setting up the sync, both Tomdroid and Tomboy fails to connect to my server.
I get a 400 Bad Request from Tomboy.
I don't see other errors in my server logs, or in OwnCloud log.
I was experiencing this issue with OwnCloud 7, and testing on another server and upgrading to OwnCloud 8 didn't solve my problem.

Maybe I'm missing something obvious...

Could you please help me ?
Tell me if you need any information.

Thank you

Edit :
I've just found this thread https://bugzilla.redhat.com/show_bug.cgi?id=1241172
Maybe it is related to a bug on CentOS...
Edit 2 : 
I've applied a temporary fix https://forum.owncloud.org/viewtopic.php?f=31&t=29185#p91250
It solves some problems I had with OwnCloud, but I still get the same error with Tomboy : 

```
[DEBUG 17:26:09.190] Building web request for URL: https://owncloud.MYWEBSITE.com/index.php/apps/grauphel/oauth/request_token
[ERROR 17:26:09.534] Caught exception. Message: The remote server returned an error: (400) Bad Request.
```

Thanks for any advice

**Comment by vasyugan** [comment]
Seeing the same on a server running Ubuntu 14.4, This is in the apache log:

```
XXX.xxx.199.221 - - [13/Sep/2015:20:13:55 +0200] "GET /index.php/apps/grauphel/api/1.0 HTTP/1.1" 200 4550 "-" "-"
XXX.xxx.199.221 - - [13/Sep/2015:20:13:56 +0200] "POST /index.php/apps/grauphel/oauth/request_token HTTP/1.1" 400 1076 "-" "-"
```

On my local raspberrypi server, running Arch Linux, it works properly. There I see the following: 

```
192.168.30.23 - - [13/Sep/2015:20:52:58 +0200] "GET /owncloud/index.php/apps/grauphel/api/1.0 HTTP/1.1" 200 335
192.168.30.23 - - [13/Sep/2015:20:52:59 +0200] "POST /owncloud/index.php/apps/grauphel/oauth/request_token HTTP/1.1" 200 96
192.168.30.23 - - [13/Sep/2015:20:53:01 +0200] "GET /owncloud/index.php/apps/grauphel/oauth/authorize?oauth_token=rda236b92bcab9ad4&oauth_callback=http%3a%2f%2flocalhost%3a8000%2ftomboy-web-sync%2f HTTP/1.1" 200 12847
192.168.30.23 - - [13/Sep/2015:20:53:03 +0200] "GET /owncloud/index.php/core/js/oc.js?v=a2d3e5e7241ed11cdce00c3b187723c3 HTTP/1.1" 200 2269
192.168.30.23 - - [13/Sep/2015:20:53:05 +0200] "GET /owncloud/index.php/avatar/jr/128?requesttoken=yWU8JLZRNJeuJ6PfjJkHOZFHnD9NYc HTTP/1.1" 200 40
192.168.30.23 - - [13/Sep/2015:20:53:05 +0200] "GET /owncloud/cron.php HTTP/1.1" 200 20
```

**Comment by cweiske** [comment]
I can't debug that from here since I don't experience the error. You could either give me access to your server (tell me the owncloud URL and login data via email to cweiske+grauphel@cweiske.de), or install wireshark and check what exactly is returned when Tomboy gets the 400 error from the server.

This information may help us to go further.

**Comment by MegaS0ra** [comment]
I've just sent you an email with login details.
Thanks a lot

**Comment by cweiske** [comment]
The "400 Bad request" response has the following content:

```
oauth_problem=parameter_absent&oauth_parameters_absent=oauth_consumer_key%26oauth_signature%26oauth_signature_method%26oauth_nonce%26oauth_timestamp
```

But the request itself _has_ them in the Authorization header:

```
Authorization:   OAuth realm="Snowy",oauth_callback="http%3A%2F%2Flocalhost%3A8000%2Ftomboy-web-sync%2F",oauth_consumer_key="anyone",oauth_nonce="2628226",oauth_signature="lP
             sb%2fiMHSZqSXFAYVVygwjYv1Cs%3d",oauth_signature_method="HMAC-SHA1",oauth_timestamp="1442663484",oauth_version="1.0"
```

**Comment by MegaS0ra** [comment]
This fixed my problem, thanks a lot !

**Comment by vasyugan** [comment]
Just for the record: I updated grauphel to 0.6 on my server and the situation remains very much unchanged. It is a server running Ubuntu 14.4, PHP 5.5.9. What's a bit special is that we run various chroots with suexec and php-fpm. I'm not sure whether that's relevant. I am currently travelling. I can try to play around with wireshark later.

**Comment by cweiske** [comment]
@vasyugan: This is _not_ fixed with version 0.6.0. 0.6.0 was released before this fix here. I'll release 0.6.1 later with this fix included.

**Comment by cweiske** [comment]
Version 0.6.1 is released which contains the fix for this issue.
